downloader/src/main/kotlin/vcs/GitWorkingTree.kt

```diff
@@ -21,68 +21,85 @@ package com.here.ort.downloader.vcs
-        return isInsideWorkTree.isSuccess && isInsideWorkTree.stdout.trimEnd().toBoolean()
+        return mutableListOf<String>().also { paths ->
+            listSubmodules("", repo, paths)
+        }
     }
 
-    override fun isShallow(): Boolean {
-        val dotGitDir = gitBase.run(workingDir, "rev-parse", "--absolute-git-dir").stdout.trimEnd()
-        return File(dotGitDir, "shallow").isFile
-    }
+    override fun getNested(): Map<String, VcsInfo> =
+        runCatching {
+            // This throws a NullPointerException if "repo" is null.
+            val paths = listSubmodulePaths(repo!!)
+            paths.associateWith { GitWorkingTree(repo.workTree.resolve(it), gitBase).getInfo() }
+        }.getOrElse {
+            emptyMap()
+        }
 
-    override fun getNested(): Map<String, VcsInfo> {
-        val root = getRootPath()
-        val paths = gitBase.run(root, "submodule", "--quiet", "foreach", "--recursive", "echo \$displaypath").stdout
-            .lines().filter { it.isNotBlank() }
+    override fun getRemoteUrl(): String =
+        runCatching {
+            // This throws a NullPointerException if "repo" is null.
+            val remotes = org.eclipse.jgit.api.Git(repo!!).remoteList().call()
+            val remoteForCurrentBranch = BranchConfig(repo.config, repo.branch).remote
 
-        return paths.associateWith { GitWorkingTree(root.resolve(it), gitBase).getInfo() }
-    }
+            val remote = if (remotes.size <= 1 || remoteForCurrentBranch == null) {
+                remotes.firstOrNull()
+            } else {
+                remotes.find { remote ->
+                    remote.name == remoteForCurrentBranch
+                }
+            }
 
-    override fun getRemoteUrl() =
-        gitBase.run(workingDir, "remote", "get-url", getRemoteForCheckout()).stdout.trimEnd()
+            remote?.urIs?.firstOrNull()?.toString()
+        }.getOrNull().orEmpty()
 
-    override fun getRevision() = gitBase.run(workingDir, "rev-parse", "HEAD").stdout.trimEnd()
+    override fun getRevision(): String = repo?.exactRef(Constants.HEAD)?.objectId?.name().orEmpty()
 
-    override fun getRootPath() =
-        File(gitBase.run(workingDir, "rev-parse", "--show-toplevel").stdout.trimEnd('\n', '/'))
+    override fun getRootPath(): File = repo?.workTree ?: workingDir
 
-    private fun listRemoteRefs(namespace: String): List<String> {
-        val tags = gitBase.run(workingDir, "-c", "credential.helper=", "-c", "core.askpass=echo", "ls-remote",
-            "--refs", getRemoteForCheckout(), "refs/$namespace/*").stdout.trimEnd()
-        return tags.lines().map {
-            it.split('\t').last().removePrefix("refs/$namespace/")
+    override fun listRemoteBranches(): List<String> =
+        runCatching {
+            LsRemoteCommand(repo).setHeads(true).call().map { it.name.removePrefix("refs/heads/") }
+        }.getOrElse {
+            emptyList()
```

Comment:
Similar to a previous comment, do we really want to suppress the exception? Maybe it could be useful to add a debug log if it failes with a Git exception. Same below.